clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -148,283 +149,188 @@ def _load_specifications(
+def _complete_examination_dates(
     rid: int,
-    exam_dates: List[Union[str, int]],
-    visit_codes: List[str],
     clinical_data_dir: Path,
-) -> List[Union[str, int]]:
-    """Clean the exam dates when necessary by trying to compute them from other sources.
-
-    Parameters
-    ----------
-    rid : int
-        Patient study/source id
-    exam_dates : List
-        Ex : ['10/12/2007', '05/29/2009', '10/25/2010', -4]
-    visit_codes : List
-        Ex : ['bl', 'm18', 'm36', 'm54']
-    clinical_data_dir : Path
-
-    Returns
-    -------
-    List of cleaned exam dates
-
-    """
-
-    from clinica.utils.stream import cprint
-
-    exam_dates_cleaned: List[str] = []
-    for visit_code, exam_date in zip(visit_codes, exam_dates):
-        if exam_date == "-4":
-            exam_date = _find_exam_date_in_other_csv_files(
-                rid, visit_code, clinical_data_dir
-            ) or _compute_exam_date_from_baseline(visit_code, exam_dates, visit_codes)
-            if not exam_date:
-                cprint(
-                    f"No EXAMDATE for subject %{rid}, at session {visit_code}",
-                    lvl="debug",
-                )
-                exam_date = "-4"
-        exam_dates_cleaned.append(exam_date)
-
-    return exam_dates_cleaned
+    session_id: Optional[str] = None,
+    examination_date: Optional[str] = None,
+) -> Optional[str]:
+    if examination_date:
+        return examination_date
+    if session_id:
+        return _find_exam_date_in_other_csv_files(rid, session_id, clinical_data_dir)
+    return None
 
 
 def _find_exam_date_in_other_csv_files(
-    rid: int, visit_code: str, clinical_data_dir: Path
+    rid: int, session_id: str, clinical_data_dir: Path
 ) -> Optional[str]:
     """Try to find an alternative exam date by searching in other CSV files."""
-    # todo (LATER) : refactor and test depending on _get_csv_files
-    for csv_file in _get_csv_files(clinical_data_dir):
-        if "aibl_flutemeta" in csv_file:
-            csv_data = pd.read_csv(
-                csv_file, low_memory=False, usecols=list(range(0, 36))
-            )
-        else:
-            csv_data = pd.read_csv(csv_file, low_memory=False)
-        exam_date = csv_data[(csv_data.RID == rid) & (csv_data.VISCODE == visit_code)]
+    from clinica.iotools.converter_utils import viscode_to_session
+
+    for csv in _get_csv_files_for_alternative_exam_date(clinical_data_dir):
+        csv_data = pd.read_csv(csv, low_memory=False)
+        csv_data["SESSION"] = csv_data.VISCODE.apply(lambda x: viscode_to_session(x))
+        exam_date = csv_data[(csv_data.RID == rid) & (csv_data.SESSION == session_id)]
         if not exam_date.empty and exam_date.iloc[0].EXAMDATE != "-4":
             return exam_date.iloc[0].EXAMDATE
     return None
 
 
-def _get_csv_files(clinical_data_dir: Path) -> List[str]:
+def _get_csv_files_for_alternative_exam_date(
+    clinical_data_dir: Path,
+) -> Generator[Path, None, None]:
```

Comment:
This returns a `Generator[Path]`, right ?

Reply:
Yes ! I had to modify the typing to have 3 arguments in brackets, following the [docs](https://docs.python.org/3/library/typing.html#typing.Generator). Else, it raises an error. But maybe I should not put None for the _input_ and _return_ fields.

Reply:
Since it's a simple iterator, we can do:

```python
def func(...) -> Iterator[Path]:
    ...
```